test/integration_tests/test_models.py

```diff
@@ -1,67 +1,56 @@
 import torch
-import torchtext
+from parameterized import parameterized
+from torchtext.models import (
+    XLMR_BASE_ENCODER,
+    XLMR_LARGE_ENCODER,
+    ROBERTA_BASE_ENCODER,
+    ROBERTA_LARGE_ENCODER,
+)
 
 from ..common.assets import get_asset_path
 from ..common.torchtext_test_case import TorchtextTestCase
 
+TEST_MODELS_PARAMETERIZED_ARGS = [
+    ("xlmr.base.output.pt", "XLMR base Model Comparison", XLMR_BASE_ENCODER),
+    ("xlmr.large.output.pt", "XLMR base Model Comparison", XLMR_LARGE_ENCODER),
+    (
+        "roberta.base.output.pt",
+        "Roberta base Model Comparison",
+        ROBERTA_BASE_ENCODER,
+    ),
+    (
+        "roberta.large.output.pt",
+        "Roberta base Model Comparison",
+        ROBERTA_LARGE_ENCODER,
+    ),
+]
 
-class TestModels(TorchtextTestCase):
-    def test_roberta_base(self):
-        asset_path = get_asset_path("roberta.base.output.pt")
-        test_text = "Roberta base Model Comparison"
-
-        roberta_base = torchtext.models.ROBERTA_BASE_ENCODER
-        transform = roberta_base.transform()
-        model = roberta_base.get_model()
-        model = model.eval()
 
-        model_input = torch.tensor(transform([test_text]))
-        actual = model(model_input)
-        expected = torch.load(asset_path)
-        torch.testing.assert_close(actual, expected)
-
-    def test_roberta_base_jit(self):
-        asset_path = get_asset_path("roberta.base.output.pt")
-        test_text = "Roberta base Model Comparison"
-
-        roberta_base = torchtext.models.ROBERTA_BASE_ENCODER
-        transform = roberta_base.transform()
-        transform_jit = torch.jit.script(transform)
-        model = roberta_base.get_model()
-        model = model.eval()
-        model_jit = torch.jit.script(model)
-
-        model_input = torch.tensor(transform_jit([test_text]))
-        actual = model_jit(model_input)
-        expected = torch.load(asset_path)
-        torch.testing.assert_close(actual, expected)
-
-    def test_roberta_large(self):
-        asset_path = get_asset_path("roberta.large.output.pt")
-        test_text = "Roberta base Model Comparison"
+class TestModels(TorchtextTestCase):
+    @parameterized.expand(TEST_MODELS_PARAMETERIZED_ARGS)
```

Comment:
I think you can go further to make the jit part parameterized as well.

You can parameterize the behavior around jitting in `test_model`, then make a Cartesian product of `TEST_MODELS_PARAMETERIZED_ARGS` and `jit=True|False` to run all the tests.

Reply:
Thanks for the feedback. Will do this in a followup PR by pulling in the `nested_params` helper method from torchaudio.